### PR TITLE
make work mist variable to $I answear state

### PR DESCRIPTION
**Pull request by alkabal**
Todo combine with Flood

Hi

If you agree i havemake a small patch for use :
        // also $G and $I
        // [G0 G54 G17 G21 G90 G94 M0 M5 M9 T0 F0.]
        stream->printf("[G%d %s G%d G%d G%d G94 M0 M%c M%c T%d F%1.4f S%1.4f]\n",

Br

### CODE REVIEW

File: src/modules/utils/simpleshell/SimpleShell.cpp [hunk 1]
```diff
@@ -870,6 +892,8 @@ void SimpleShell::get_command( string parameters, StreamOutput *stream)
             THEROBOT->inch_mode ? 20 : 21,
             THEROBOT->absolute_mode ? 90 : 91,
             get_spindle_state()?'3':'5',
+            get_mist_state()?'7':'9',
+//            get_flood_state()?'8':'9',     // TODO How to combine in the same printf var ???
```
Comment (wolfmanjm) [inline src/modules/utils/simpleshell/SimpleShell.cpp]: is it always either M7 or M8 (or M9)

**Comment by alkabal** [comment]
Yes I think 

M7 mist on
M8 flood on
M9 mist flood off


Br

File: src/modules/utils/simpleshell/SimpleShell.cpp (hunk 1 above)
Comment (alkabal) [reply src/modules/utils/simpleshell/SimpleShell.cpp]: From Linuxcnc/GRBL yes.

I haven't updated Gcode parser, only the data variable.

This data is triggered with the switch.mist.enable without other code change.

Br

**Comment by wolfmanjm** [comment]
Then you should do this...

```
get_flood_state() ? '8' : get_mist_state() ? '7' : '9';
```

**Comment by alkabal** [comment]
Hi

Perfect with your code line thanks a lot !

I have too early edited the wiki http://smoothieware.org/supported-g-codes
I hope you not angry with this or i can revert the change immediatly (sorry)

Br

**Pull request merged**